tests/conftest.py

```diff
@@ -149,103 +179,228 @@ def test_something(remote: Remote):
+        # avoid test hanging on 2FA prompt.
+        pytest.skip(reason=f"Test needs an existing connection to {cluster} to run.")
+    return get_slurm_account(cluster)
 
-    if not logged_in_before_tests and should_already_be_logged_in_during_tests:
-        console.log(
-            f"No prior connection to {cluster} before running tests, creating one."
-        )
 
-        if cluster in DRAC_CLUSTERS:
-            # todo: Seems like logger.warning is not being displayed in the test output
-            # somehow?
-            console.log(
-                f"Going through 2FA with cluster {cluster} only once before tests."
-            )
+@pytest.fixture(scope="session")
+def max_job_duration(
+    request: pytest.FixtureRequest, cluster: str
+) -> datetime.timedelta:
+    """Fixture that allows test to parametrize the duration of their jobs."""
+    return getattr(request, "param", MAX_JOB_DURATION)
 
-        RemoteV2(cluster)
-        assert is_already_logged_in(cluster, also_run_command_to_check=True)
-        yield True
-        # TODO: Should we remove the connection after running the tests?
-        return
 
-    control_path = get_controlpath_for(cluster)
-    assert control_path.exists()
-    backup_dir = tmp_path_factory.mktemp("backup")
+@pytest.fixture(scope="session")
+def allocation_flags(
+    request: pytest.FixtureRequest,
+    slurm_account_on_cluster: str,
+    job_name: str | None,
+    max_job_duration: datetime.timedelta,
+) -> list[str]:
+    """Flags passed to salloc or sbatch during tests.
 
-    console.log(
-        f"Temporarily moving {control_path} to {backup_dir} so tests are run without "
-        f"an existing connection."
+    When parametrized, overrides individual flags:
+    ```python
+    @pytest.mark.parametrize("allocation_flags", [{"some_flag": "some_value"}], indirect=True)
+    def some_test(allocation_flags: list[str])
+        assert "--some_flag=some_value" in allocation_flags
+    ```
+    """
+    default_allocation_options = {
+        "wckey": WCKEY,
+        "account": slurm_account_on_cluster,
+        "nodes": 1,
+        "ntasks": 1,
+        "cpus-per-task": 1,
+        "mem": "1G",
+        "time": max_job_duration,
+        "oversubscribe": None,  # allow multiple such jobs to share resources.
+    }
+    if job_name is not None:
+        # Only set the job name when needed. For example, `mila code` tests don't want
+        # it to be set.
+        default_allocation_options["job-name"] = job_name
+    overrides = getattr(request, "param", {})
+    assert isinstance(overrides, dict)
+    if overrides:
+        print(f"Overriding allocation options with {overrides}")
+        default_allocation_options = default_allocation_options.copy()
+        default_allocation_options.update(overrides)
+    return [
+        f"--{key}={value}" if value is not None else f"--{key}"
+        for key, value in default_allocation_options.items()
+    ]
+
+
+@pytest.fixture()
+def ssh_config_file(
```

Comment:
Sorry @satyaog ,  I'm not sure I understand what you mean.. Perhaps I can clarify what I meant to do here, and then you could help me understand your question better?
Here's some context, in case it's helpful: Here my goal is to run tests with the same SSH config that users would get after running `mila init`. Instead of using a static SSH config file that would be hard-coded into the tests, I decided to use the function to generate the file (which is also what's done in all the regression tests for `mila init`).

Reply:
Well, as I mentioned, the file is itself "fixed" because of the regression tests for `mila init`. If I were to use a fixed file, tests using this fixture could eventually become out-of-sync as new changes are made to the ssh config by `mila init`.
I agree that the testing code could be simpler and easier to read, with fewer moving parts. I'll keep thinking about this for the next few PRs.

Reply:
> Here my goal is to run tests with the same SSH config that users would get after running mila init. Instead of using a static SSH config file that would be hard-coded into the tests, I decided to use the function to generate the file (which is also what's done in all the regression tests for mila init).

This makes me think that `ssh` proper supports:

  - The `-F configfile` for specifying an alternate config file and
  - The `-G` flag to cause `ssh` to parse the config file and arguments, then dump out the final configuration values for the connection instead of attempting to use them.

That could be used to copy the config file, patch it on the side, have it be parsed, and you could assert that the important keys are set to the correct values.